pkg/api/v1/routes/handlers.go

```diff
@@ -251,40 +224,44 @@ func (r *Routes) serverConditionCreate(c *gin.Context) {
 	// check if any condition with exclusive set is in incomplete states
 	if errEx := r.exclusiveNonFinalConditionExists(c.Request.Context(), serverID); errEx != nil {
 		if errors.Is(errEx, ErrConditionExclusive) {
-			c.JSON(
-				http.StatusBadRequest,
-				&v1types.ServerResponse{Message: errEx.Error()},
-			)
+			r.logger.WithFields(logrus.Fields{
+				"error": err,
+				// XXX: would be nice to have the id of the exclusive condition here too
+			}).Info("active, exclusive condition on this server")
 
-			return
+			return http.StatusBadRequest, &v1types.ServerResponse{
+				Message: errEx.Error(),
+			}
 		}
 
-		c.JSON(
-			http.StatusInternalServerError,
-			&v1types.ServerResponse{Message: errEx.Error()},
-		)
+		r.logger.WithFields(logrus.Fields{
+			"error": err,
+		}).Info("error checking for exclusive conditions")
 
-		return
+		return http.StatusInternalServerError, &v1types.ServerResponse{
+			Message: errEx.Error(),
+		}
 	}
 
 	// Create the new condition
 	err = r.repository.Create(c.Request.Context(), serverID, condition)
 	if err != nil {
-		c.JSON(
-			http.StatusInternalServerError,
-			&v1types.ServerResponse{Message: err.Error()},
-		)
 		r.logger.WithFields(logrus.Fields{
 			"error": err,
 		}).Info("condition create failed")
 
-		return
+		return http.StatusInternalServerError, &v1types.ServerResponse{
+			Message: err.Error(),
+		}
 	}
 
+	// XXX: this operation can't ignore errors
```

Comment:
My thoughts here was that the Condition `Orchestrator` component could follow up on `Condition`s that have no updates in the last X time period...

Reply:
I filed [issue 12](https://github.com/metal-toolbox/conditionorc/issues/12) to address this. This could be a simple "watcher" go-routine, or something more elaborate.